instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -65,17 +67,31 @@ public static class OpenTelemetrySdkConfig {
 
     @Bean
     @ConfigurationPropertiesBinding
-    @ConditionalOnBean({
-      OtelResourceAutoConfiguration.class,
-      OtlpLoggerExporterAutoConfiguration.class,
-      OtlpSpanExporterAutoConfiguration.class,
-      OtlpMetricExporterAutoConfiguration.class
-    })
+    @Conditional(MapConverterCondition.class)
     public MapConverter mapConverter() {
       // needed for otlp exporter headers and OtelResourceProperties
       return new MapConverter();
     }
 
+    static final class MapConverterCondition implements Condition {
+
+      private static final List<Class<?>> ANY_REQUIRED_BEANS =
+          Arrays.asList(
+              OtelResourceAutoConfiguration.class,
+              OtlpLoggerExporterAutoConfiguration.class,
+              OtlpSpanExporterAutoConfiguration.class,
+              OtlpMetricExporterAutoConfiguration.class);
+
+      @Override
+      public boolean matches(
+          org.springframework.context.annotation.ConditionContext context,
+          org.springframework.core.type.AnnotatedTypeMetadata metadata) {
+
+        return ANY_REQUIRED_BEANS.stream()
+            .anyMatch(s -> context.getBeanFactory().getBeanProvider(s).getIfAvailable() != null);
```

Comment:
you sure this is safe? As far as I understand this creates the actual bean instance while https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/condition/OnBeanCondition.java doesn't

Reply:
I suspect that when you create the beans too early you could run into something like https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9988 I pasted the wrong like into my previous comment (update it) as a sample of how this might be done https://www.baeldung.com/spring-conditional-annotations#combining-conditions I have not tried it myself so no idea whether this would actually work. I was thinking that if you could have `@ConditionalOnBean` on nested classes then perhaps this would have less chances of breaking in a strange way.

Reply:
learned something new :smile: 

That's indeed a good argument to use the nested conditions - done!